### PR TITLE
{Bounty: Paid} Virus Symptoms with Positive Thresholds Suck Slightly Less

### PR DESCRIPTION
**Pull request by DatBoiTim**
## About The Pull Request
Bounty provided by LeeStriter.
So there are a few pre-existing symptoms that some people raise issue with in context of positive thresholds.

**Bee Infestation:**

With the positive threshold, the bees spawned inject honey. This is a problem, as they can eventually coma others. Now it spawns a honey bee and an insulin bee. While it doesn't completely negate the risk, it certainly minimizes the effect to some extent.

**DNA Saboteur:**

Positive Threshold used to include some not very enjoyable mutations. These are no more. However, not a free hulk/tk/xray/etc

**Biometallic Replication:**

If you have the beneficial threshold, your brain is lubricated, so no more _**CLANG CLANG CLANG**_ every 5 seconds. This still exists if you don't have the positive threshold.

**Pituitary Disruption:**

Plasma filter can no longer randomly happen to non-plasmemes, resulting in fewer deaths from this disease. Still does cell damage, requiring Organic Flux, but hey it sucks slightly less.

**Viral Aggressive Metabolism**

When it self cures, resistance isn't added, so you can be reinfected. This allows for "virological potions" that you get infected with, and have active symptoms take effect immediately.

## Why It's Good For The Game
A few of the "beneficial thresholds" on symptoms have had some very undesirable if not harmful side effects. Sound spam is annoying, RNG organ replacement is annoying, genetic mutations that don't match expectations, or bees that give the entire station diabetes are all very unfun.

This makes this better. It also makes a useless symptom a smidge better.

## Changelog
:cl: DatBoiTim
tweak: Bee Viro Symptom spawns insulin bees alongside honey bees on positive threshold
tweak: DNA Saboteur no longer gives neutral mutations on positive threshold
tweak: Biometallic Replication no longer makes that annoying clanging sound every five seconds on positive threshold
tweak: Pituitary Disruption won't give you plasmeme lungs anymore
tweak: Viral Aggressive Metabolism No Longer Vaccinates When Self Curing
/:cl:

### CODE REVIEW

**Comment by leestriter** [comment]
Hell yeah, you did this way faster them expected!

**Comment by utilbird** [comment]
Why do you always put bounty payment status in the title 🤔

**Comment by DatBoiTim** [comment]
> Why do you always put bounty payment status in the title 🤔 

2 reasons
- These are privately negotiated and help build credibility for the person who gives the bounty.
- These are bounties that are paid on PR, regardless of merge. So preferably, you guys would only key when it's paid.

**Comment by utilbird** [comment]
Fair

File: code/datums/diseases/advance/symptoms/clockwork.dm
Comment (utilbird) [inline code/datums/diseases/advance/symptoms/clockwork.dm]:
```suggestion
	if(prob(5) && !robust)
		SEND_SOUND(owner, pickweight(list('sound/effects/clock_tick.ogg' = 6, 'sound/effects/smoke.ogg' = 2, 'sound/spookoween/chain_rattling.ogg' = 1, 'sound/ambience/ambiruin3.ogg' = 1)))
```

File: code/datums/diseases/advance/symptoms/viral.dm
Comment (zeskorion) [inline code/datums/diseases/advance/symptoms/viral.dm]: keep this at -3 or else it can potentially be used for very nasty death plagues. if you wanna buff it, make it not vaccinate on cure

**Review by zeskorion — CHANGES_REQUESTED** [review]

File: code/datums/diseases/advance/symptoms/viral.dm
Comment (DatBoiTim) [reply code/datums/diseases/advance/symptoms/viral.dm]: Sorry chief this was requested for the bounty. I didn't pull this idea myself.

Comment (zeskorion) [reply code/datums/diseases/advance/symptoms/viral.dm]: iirc -3 was the minimum to not be antag fuel, which is why it was removed the first time. making it not vaccinate on cure lets you make virus "potions" and stuff

File: code/datums/diseases/advance/symptoms/clockwork.dm
Comment (DatBoiTim) [reply code/datums/diseases/advance/symptoms/clockwork.dm]: Committed

File: code/datums/diseases/advance/symptoms/viral.dm
Comment (DatBoiTim) [reply code/datums/diseases/advance/symptoms/viral.dm]: I can look at that then.

Comment (leestriter) [reply code/datums/diseases/advance/symptoms/viral.dm]: What are those death plauges? The level 9 death plague symptoms have - 2 transmission already. With the negative 2 from viral aggressive, if you use the best transmission symptoms you can only get to an 8 transmission, so it'd still only be blood transmission and not contagious. None of what zesko is saying is backed up. -2 + -2 +4 +3 + 3 + 2 is 8. And even for that, the resistance and stage speed would suck.

Comment (leestriter) [reply code/datums/diseases/advance/symptoms/viral.dm]: > iirc -3 was the minimum to not be antag fuel, which is why it was removed the first time. making it not vaccinate on cure lets you make virus "potions" and stuff

It does vaccinate on cure though, and none of the virus "potions" you could make are even any good if it didn't.

Comment (leestriter) [reply code/datums/diseases/advance/symptoms/viral.dm]: > keep this at -3 or else it can potentially be used for very nasty death plagues. if you wanna buff it, make it not vaccinate on cure

Show me the death plague this change enables and I'll give datboytim permission to remove that change. I paid for this PR, so I'm the one to convince. And I won't be convinced by baseless claims.

If I'm not convinced and datboytim wants to remove that from the pr, he can pay me back and it'll be his pr.

Comment (DatBoiTim) [reply code/datums/diseases/advance/symptoms/viral.dm]: So I will be honest, not vaccinating sounds way more interesting than standard buffing symptom that kinda sucks.

It's hard to imagine what precisely could be done but man I'd rather go for something cool than "Oh boy, number go brrrr"

Comment (leestriter) [reply code/datums/diseases/advance/symptoms/viral.dm:76]: I'll be completely honest. The reason for changing it is entirely to enable a trans 12 viral aggressive metabolism macrophage, so I could use macrophage to spread positive low transmission illnesses like exo loco used to do, without causing wide spread destruction from long term macrophage. If it enables a death plague we can sttle for non vaccinating, though non vaccinating might buff death plagues more then the + 1 transmission.

Comment (leestriter) [reply code/datums/diseases/advance/symptoms/viral.dm]: I'll be completely honest. The reason for changing it is entirely to enable a trans 12 viral aggressive metabolism macrophage, so I could use macrophage to spread positive low transmission illnesses like exo loco used to do (specfiically all thresholds superficial healing and regen coma), without causing wide spread destruction from long term macrophage. If it enables a death plague we can settle for non vaccinating, though non vaccinating might buff death plagues more then the + 1 transmission. You have to neuter the disease with spacecillin of course.

Comment (zeskorion) [reply code/datums/diseases/advance/symptoms/viral.dm]: Not only is macrophage decently deadly in this form, and will attack once the disease is cured by VAM, it can be used to spread such a death plague with VAM on it- which with things like ARDS is incredibly tough to cure before dying. 

VAM is intended to, by design, be nearly impossible to spread on its own. It's for single-target diseases, not things you spread

Comment (leestriter) [reply code/datums/diseases/advance/symptoms/viral.dm]: Macrophage isn't really that deadly, and not at all paired with regen coma and superficial healing (and the macrophage having a lowish stage speed). And VAM already works with a lot of different contagious illnesses as is. Autophage zombies, and anything wiith a transmission of at least 0 can be paired with it and spread via touch.

A macrophage spread death plague is far worse without aggressive metabolism, since you can pump stage speed, or use necrotic metabolism.

Comment (leestriter) [reply code/datums/diseases/advance/symptoms/viral.dm]: It would only be a vector for use on LRP, admittedly. The monkeys I tested this on all survived.

Comment (zeskorion) [reply code/datums/diseases/advance/symptoms/viral.dm]: Look, macrophage is not intended as a vector for beneficials. DatBoi may end up making one such thing in the future if he continues keeping up my torch, but macrophage is still self-antag and against viro policy, at least if rules haven't gotten dumb. You cant do this as a nonantag anyways. VAM should not be compatable with transmission plagues that are worth anything. 

Also: said zombie virus doesnt spread itself unless something is terribly wrong with the code. Viromerol does not spread when the zombies attack.

Comment (leestriter) [reply code/datums/diseases/advance/symptoms/viral.dm]: The rules have changed substantially for viro.

Comment (leestriter) [reply code/datums/diseases/advance/symptoms/viral.dm]: The zombie virus is not acting as intended in that case, it does spread via attack.

Comment (leestriter) [reply code/datums/diseases/advance/symptoms/viral.dm]: And what was or was not originally intended by the code is irrelevant.

Comment (zeskorion) [reply code/datums/diseases/advance/symptoms/viral.dm]: Zombies do spread via attack but the virus itself doesnt. 

And no, my original intent is not irrelevant if I revamped an entire department in a cohesive and balanced manner. Its a bunch of moving cogs that fit together for a reason. 

As it stands, i'd advise maintainers not to merge this pr in this state

Comment (leestriter) [reply code/datums/diseases/advance/symptoms/viral.dm]: It is though, you left, now viro is rebalanced by others. By that logic no code should ever be changed except by its original creator. I paid for this pr, and with the attitude you are giving me, I will not have it changed. It will be merged this way or not merged.

Comment (leestriter) [reply code/datums/diseases/advance/symptoms/viral.dm]: If you want it to be your pr, pay datboitim 15 bucks, and he can pay me back, and it'll be your pr. Until then, this isn't you pr Zesko don't act like you have final say. I was hearing you out and listening to your arguments. And I wouldn't say viro is balanced or cohesive just cause you made it, people already changed much of what you made proving it was neither.

Comment (leestriter) [reply code/datums/diseases/advance/symptoms/viral.dm]: If you had provided this death plague you said was why the pr had to be changed, I would've changed it. As is, you are fumbling making up one reason after another when it really comes down to it not being your idea or original vision and you want to maintain control. Well, you shouldn't have quit in that case.

Comment (zeskorion) [reply code/datums/diseases/advance/symptoms/viral.dm]: Omigod you're actually getting pissy with me cuz i decided to come check the github just as a bunch of viro stuff was getting done?
I've checked.  There have been no merged viro prs since i left. Just because i am not here does not mean my opinion does not hold weight

No idea why you seem to hate the fuck out of me now. But given that shit hasn't changed, my opinion is still valid. Im not here to flamewar though. Just offer my two cents, cuz i know what im talking about

Anyways, i gotta get some sleep. Toodles~

Comment (leestriter) [reply code/datums/diseases/advance/symptoms/viral.dm]: I don't hate you, you overstepped on this pr, and that's it. I am glad you have come back, but you came in and acted like you have final say on what gets merged. And they have removed exo loco and made other changes since you left.

Comment (zeskorion) [reply code/datums/diseases/advance/symptoms/viral.dm]: Also: i just woke up im not designing any diseases, im about to go back to sleep

And no one has changed anything, unless you count austation's viro buff by lostman, which actively makes it overpowered

Viro is in a nice, cozy niche where it's powerful enough if you're good, okayish if not, but it has a hard power ceiling that makes it hard to game. If viro WERE gameable, it would very quickly annoy people due to its nature. And then it would get nerfed,  because of its nature.

Viro was nerfed in that way once. Twice, counting what i did to the major death plagues. Do you want that happening again?

Comment (zeskorion) [reply code/datums/diseases/advance/symptoms/viral.dm]: Anyways, brain stupid doesnt like coming back to old code shit yadda yadda dysphoria ill go to bed. Have fun.

Comment (leestriter) [reply code/datums/diseases/advance/symptoms/viral.dm]: That has literally nothing to do with buffing VAMs transmission by 1. You are changing the subject. And you are one who said not to change it because "keep this at -3 or else it can potentially be used for very nasty death plagues. if you wanna buff it, make it not vaccinate on cure". All I asked was what that disease you are referencing is, and I showed the math for why that isn't true.

Comment (zeskorion) [reply code/datums/diseases/advance/symptoms/viral.dm]: Oh- exolocomotive was removed before i left. There were no other big changes iirc.  I had a breakdown before i could replace it. Like i said, i checked. It has to be replaced. But macrophage doesnt work in its niche.

Comment (zeskorion) [reply code/datums/diseases/advance/symptoms/viral.dm]: Anyways.  Paying for a pr wont get it automatically merged. Making a 12 transmission macrophage is already a potentially nasty death plague- it instantly starts spreading whatever nasty single target disease you can come up with, which can include another VAM disease

Comment (leestriter) [reply code/datums/diseases/advance/symptoms/viral.dm]: I'm not saying its great in that niche, but we both agree that niche needs to be filled. This is the easiest way with minimum changes to the game. I'm not gonna pretend that macrophage spread for a good diseases is great. I just don't have anything else.

Comment (leestriter) [reply code/datums/diseases/advance/symptoms/viral.dm]: So as for arguing about balance, I just want to add back an old ability we both know viro needed.

Comment (zeskorion) [reply code/datums/diseases/advance/symptoms/viral.dm]: God, dm me on discord or something. I dont wanna argue over github opened via my email in this shitty interface, spamming poor tim.

This is pointless

Comment (leestriter) [reply code/datums/diseases/advance/symptoms/viral.dm]: We have talked on discord, and I am going with zesko's suggestion of making it not vaccinate on cure and not the transmission change.

Comment (DatBoiTim) [reply code/datums/diseases/advance/symptoms/viral.dm]: Too late for RIP Inbox.

As Zesko has said, I am carrying the torch. This means a xenolocomotive alternative is going to happen. Lee, I understand your buff was to enable a symptom to function similar to it, however, with a better vector coming in the pipeline, I feel it is safe to say that making macrophage a vector is not really the best approach for the purposes mentioned.

To my understanding, you both had come to an agreement as well. So we will call the issue of no-vax vs. transmission buff settled by both pipeline of a less physically aggressive vector and [Rule Of Cool](https://tvtropes.org/pmwiki/pmwiki.php/Main/RuleOfCool).

Code Resolution will happen soon.

Comment (utilbird) [reply code/datums/diseases/advance/symptoms/viral.dm]: > If you want it to be your pr, pay datboitim 15 bucks, and he can pay me back, and it'll be your pr. Until then, this isn't you pr Zesko don't act like you have final say.

Pull your shit together dude. Zesko was voicing his concern in a respectful manner. I don't care who paid for the PR because it doesn't change anything, money does not add any weight to a balance decision.

Comment (leestriter) [reply code/datums/diseases/advance/symptoms/viral.dm]: I am sorry, I overreacted.

Comment (leestriter) [reply code/datums/diseases/advance/symptoms/viral.dm]: I was trying to understand his concerns and he just says, people should not merge this and I got upset, but what I said wasn't justified.

Comment (DatBoiTim) [reply code/datums/diseases/advance/symptoms/viral.dm]: I'll be resolving this to move forward with the agreed development path.

File: code/datums/diseases/advance/symptoms/heal.dm
Comment (utilbird) [inline code/datums/diseases/advance/symptoms/heal.dm:516]: Hmm, wouldn't it be better to just not have the disease add them in the first place?

Comment (DatBoiTim) [reply code/datums/diseases/advance/symptoms/heal.dm:516]: That's what it's doing. Its removing the incorrect lungs from the list of possible organs

Comment (DatBoiTim) [reply code/datums/diseases/advance/symptoms/heal.dm:516]: Additionally taking that into the loop means those checks are running every single time, whereas this just nukes them from it once.

**Review by utilbird — APPROVED** [review]

**Pull request merged**
